pkg/scheduler/scheduler.go

```diff
@@ -372,7 +372,7 @@ const maxScheduleRetryCount = 3
 const maxScheduleRetryDuration = 10 * time.Minute
 
 func (s *Scheduler) addRequestToBacklog(request *types.ContainerRequest) error {
-	if request.Gpu != "" && request.GpuCount <= 0 {
+	if len(request.GpuRequest) > 0 && request.GpuCount <= 0 {
```

Comment:
how do you feel about adding a struct function to ContainerRequest like RequiresGPU()
which inspects r.GpuRequest, for example:

```
func (r *ContainerRequest) RequiresGPU() bool {
   return len(r.GpuRequest) > 0
}
```

Reply:
could clean up quite a few places i think?

Reply:
yea i think thats a good idea